### PR TITLE
Ts/1925 funman workflow

### PR DESCRIPTION
**Pull request by Tom-Szendrey**
Initial funman exploration. Sorry for the massive PR
A few things will be done in future PRs such as:

-Poller to follow regular format for getStatus #2196
-cleanup of UI #1925
-Storing this in TDS as object #1890
-Update renderer to display false boxes for current and future timesteps to more accurately display to the user #1923

### CODE REVIEW

File: packages/client/hmi-client/src/components/funman/tera-constraint-group-form.vue
Comment (YohannParis) [inline packages/client/hmi-client/src/components/funman/tera-constraint-group-form.vue:1]: Who wrote this abomination!?

Comment (Tom-Szendrey) [reply packages/client/hmi-client/src/components/funman/tera-constraint-group-form.vue:1]: Ill never tell

Comment (mwdchang) [reply packages/client/hmi-client/src/components/funman/tera-constraint-group-form.vue:1]: Are there any actual change requests or constructive comments here?

Comment (YohannParis) [reply packages/client/hmi-client/src/components/funman/tera-constraint-group-form.vue:1]: I'm waiting for the PR to be off Draft to write constructive comments.

**Comment by dvince2** [comment]
back end 👍

File: packages/client/hmi-client/src/workflow/ops/funman/funman-operation.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+	weights?: number[]; // 1 to 1 mapping with variables
+	timepoints?: FunmanInterval;
+	interval?: FunmanInterval;
+}
+
+export interface FunmanOperationState {
+	currentTimespan: TimeSpan;
+	numSteps: number;
+	tolerance: number;
+	constraintGroups: ConstraintGroup[]; // This should be optional not required.
```
Comment (Tom-Szendrey) [inline packages/client/hmi-client/src/workflow/ops/funman/funman-operation.ts:17]: FYI I am fixing this at the moment

File: packages/client/hmi-client/src/workflow/ops/funman/tera-funman.vue [hunk 2]
```diff
@@ -0,0 +1,436 @@
...
+// Grab startTime, endTime, numberOfSteps and create list.
+function getStepList() {
+	const aList = [startTime.value];
+	const stepSize = (endTime.value - startTime.value) / numberOfSteps.value;
+	for (let i = 1; i < numberOfSteps.value; i++) {
+		aList[i] = i * stepSize;
+	}
+	aList.push(endTime.value);
+	return aList;
+}
```
Comment (mwdchang) [inline packages/client/hmi-client/src/workflow/ops/funman/tera-funman.vue:302]: This algorithm will  allow for steps that are not integers, is that okay with Funman or will we need to round/floor/ceil the numbers?

Comment (Tom-Szendrey) [reply packages/client/hmi-client/src/workflow/ops/funman/tera-funman.vue:302]: Good catch! I will look into this

File: packages/client/hmi-client/src/workflow/ops/funman/tera-funman.vue [hunk 3]
```diff
@@ -0,0 +1,436 @@
...
+
+						return {
+							name: ele.id,
+							interval: {
+								lb: ele.value,
+								ub: ele.value
+							},
+							label: 'any'
+						};
+					});
```
Comment (mwdchang) [inline packages/client/hmi-client/src/workflow/ops/funman/tera-funman.vue:344]: - Have we checked that Funman can even process observables?
- I would recommend that we move the parsing out into its own function, makes things easier to read and to do tests.

File: packages/client/hmi-client/src/workflow/ops/funman/tera-funman.vue (hunk 2 above)
Comment (Tom-Szendrey) [reply packages/client/hmi-client/src/workflow/ops/funman/tera-funman.vue:302]: timestep will be integer only

startTime, numberOfSteps, endTime are restricted to integers by the UI however the type is `number`
adding floor to stepSize

File: packages/client/hmi-client/src/workflow/ops/funman/tera-funman.vue (hunk 3 above)
Comment (Tom-Szendrey) [reply packages/client/hmi-client/src/workflow/ops/funman/tera-funman.vue]: good catch on observables.

Good call to throw this into helpers I have made
setModelOptions, setRequestParameters

**Review by mwdchang — APPROVED** [review]

**Comment by YohannParis** [comment]
Okay, I will make my PR to fix most of the front-end issues

**Pull request merged**